cmake: Remove global include of board dir

galak opened the pull request:
Remove having ${BOARD_DIR} in the global include path for Zephyr builds.
Needed to add this to the arch/posix because of how posix "boards"
define various things like interrupt handling.  Also fixed up up_squared
to build with this change.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
cmake: Remove global include of board dir

Remove having ${BOARD_DIR} in the global include path for Zephyr builds.
Needed to add this to the arch/posix because of how posix "boards"
define various things like interrupt handling.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif commented:
Please split into multiple commits

nashif commented on boards/x86/up_squared/dts_fixup.h:
```diff
@@ -7,9 +7,153 @@
 /* Board level DTS fixup file */
 
 #ifdef CONFIG_SBL_FIXUP
```
can we please put this in a separate commit?

galak replied:
Sure, separate PR?

nashif replied:
no need for new PR

galak replied:
Done

galak committed:
boards: up_squared: Rework dts_fixup.h to work w/o BOARD_DIR include

Merge up_squared_dts_fixup.h and up_squared_sbl_dts_fixup.h into
dts_fixup.h.  This allows us to remove BOARD_DIR from being included
everywhere.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

aescolar reviewed (approved):
+1 for the board include part.
